### PR TITLE
Enhance Data Table Sorting Functionality

### PR DESCRIPTION
**Pull request by joshri**
Closes: #1467

`sortWithType` function allows us to sort data table columns no matter what we're putting inside the table cells. Right now we have sorts for `string`, `number`, `date`, and `bool`. Tested in `npm run storybook` and `jest /DataTable.test.tsx`

### CODE REVIEW

File: ui/components/DataTable.tsx [hunk 1]
```diff
@@ -16,19 +16,35 @@ import Link from "./Link";
...
     value: string | ((k: any) => string | JSX.Element);
+    sortType?: SortType;
+    altSortValue?: (k: any) => any;
   }[];
   /** A list of data that will be iterated through to create the columns described in `fields`. */
   rows: any[];
-  /** A list of strings representing the sortable columns of the table, passed into lodash's `_.sortBy`. Must be lowercase. */
-  sortFields?: string[];
+  /** field to initially sort against. */
+  defaultSort: {
```
[the remark this answers is not in the document]
Comment (joshri) [reply ui/components/DataTable.tsx:42]: it ends up just being one of the fields - if I do the name I'd have to search through the fields array to find the right field right? If I do the index of the correct field as the prop then it's more confusing to use but easier to implement. What do you think?

File: ui/components/__tests__/__snapshots__/DataTable.test.tsx.snap [hunk 2]
```diff
@@ -201,18 +201,144 @@ exports[`DataTable snapshots renders 1`] = `
             className="MuiTableCell-root MuiTableCell-head"
             scope="col"
           >
-            <h2>
-              Status
-            </h2>
+            <div
+              className="c1"
+            >
+              <button
```
[the remark this answers is not in the document]
Comment (joshri) [reply ui/components/__tests__/__snapshots__/DataTable.test.tsx.snap:207]: All of our buttons are the MUI button which has some defaults for the event functions if that's what you're talking about - I can't find `a11y` in the snap so I'm a little lost haha

File: ui/components/DataTable.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpellizzari) [reply ui/components/DataTable.tsx:42]: > if I do the name I'd have to search through the fields array to find the right field right

Yes that is how I would do it. This prevents the user from having to duplicate the same sorting logic in two places.

File: ui/components/__tests__/__snapshots__/DataTable.test.tsx.snap (hunk 2 above)
[the remark this answers is not in the document]
Comment (jpellizzari) [reply ui/components/__tests__/__snapshots__/DataTable.test.tsx.snap:207]: https://developer.mozilla.org/en-US/docs/Web/Accessibility

[the remark this answers is not in the document]
Comment (joshri) [reply ui/components/__tests__/__snapshots__/DataTable.test.tsx.snap:207]: Looks like for MUI we have to add aria-labels

**Review by jpellizzari — APPROVED** [review]
Code LGTM. Nice work.

**Pull request merged**